Core/src/org/sleuthkit/autopsy/communications/AccountNode.java

```diff
@@ -88,20 +97,32 @@ protected Sheet createSheet() {
             s.put(properties);
         }
 
+        // RAMAN TBD: need to figure out how to get the right filters here
+        // We talked about either creating a wrapper class around AccountDeviceInstance to push the selected filters
+        // Or some kind of static access to pull the selected filters
```

Comment:
What if we passed the filter into the node when it was created, then it could use the same filter to do query for the count

Reply:
@millmanorama, @bcarrier has been advocating pushing the filters through as far as needed. Passing the filter through the nodes directly is a nice alternative to having a key class (@Raman's wrapper class in his comment), since it reduces the total number of abstractions. The idea behind the key class was that we would need to pair a AccountDeviceInstance with either a count of relationships or a filter, depending on where we do the relationships count query.  I suppose we could pass the counts through the node as well. Personally, I'm inclined to think we will end up passing the filter, whether in a wrapper or as a node property. The idea of the key cl;ass seems to be running out of steam.